src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -552,35 +566,68 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		this.doChangeKernel(displayName, true);
 	}
 
-	public async doChangeKernel(displayName: string, mustSetProvider: boolean = true): Promise<void> {
-		if (mustSetProvider) {
-			await this.setProviderIdAndStartSession(displayName);
+	private async doChangeKernel(displayName: string, mustSetProvider: boolean = true, restoreOnFail: boolean = true): Promise<void> {
+		let oldDisplayName = this._activeClientSession && this._activeClientSession.kernel ? this._activeClientSession.kernel.name : undefined;
+		try {
+			let changeKernelNeeded = true;
+			if (mustSetProvider) {
+				changeKernelNeeded = await this.setProviderIdAndStartSession(displayName);
+			}
+			if (changeKernelNeeded) {
+				let spec = this.findSpec(displayName);
+				if (this._activeClientSession && this._activeClientSession.isReady) {
+					let kernel = await this._activeClientSession.changeKernel(spec, this._oldKernel);
+					try {
+						await kernel.ready;
+						await this.updateKernelInfoOnKernelChange(kernel);
+					} catch (err2) {
+						// TODO should we handle this in any way?
```

Comment:
Do you think console.log will help us for troubleshooting?

Reply:
Will add logging.